### PR TITLE
essync configuration parameters.

### PR DESCRIPTION
**Pull request by mhilton**
Added a -settings and a -mapping flag to essync to
allow the user to configure the elasticsearch index correctly
before syncing the data. This saves manually setting up the server.

### CODE REVIEW

**Comment by mhilton** [comment]
This is a small update to the essync tool to make it easier to sync into new indexes. To QA run `essync -settings=index.json -mapping=entity.json` and check that the indexes are created correctly.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results: http://ci.jujugui.org:8080/job/charmstore/388/

**Comment by jrwren** [comment]
:+1:

File: cmd/essync/main.go [hunk 1]
```diff
@@ -53,5 +57,55 @@ func populate(confPath string) error {
 	if err != nil {
 		return err
 	}
+	if *settings != "" {
+		err = writeSettings(es, *index, *settings)
+		if err != nil {
```
Comment (rogpeppe) [inline cmd/essync/main.go:62]: FWIW when there are no required side effects, and all else being equal,
I'd prefer to keep err variables local. It means there's less to look
for when scanning the code. In this case (and others further down)
I'd probably use the "if err := ...; err != nil {" idiom.

Comment (mhilton) [inline cmd/essync/main.go:62]: I'm still getting used to that one. I've always done it this way in the past, but I've no strong opinion on it.

**Pull request closed without merging**